File: flutter_app/lib/models/herb.dart
class Herb {
  final int id;
  final String name;
  final String? aliasName;
  final String? pinyin;
  final int stockG;
  final String unit;
  final int warnThresholdG;
  final bool enabled;
  final String createdAt;

  Herb({
    required this.id,
    required this.name,
    this.aliasName,
    this.pinyin,
    required this.stockG,
    required this.unit,
    required this.warnThresholdG,
    required this.enabled,
    required this.createdAt,
  });

  factory Herb.fromJson(Map<String, dynamic> json) => Herb(
    id: json['id'],
    name: json['name'],
    aliasName: json['aliasName'],
    pinyin: json['pinyin'],
    stockG: json['stockG'],
    unit: json['unit'],
    warnThresholdG: json['warnThresholdG'],
    enabled: json['enabled'],
    createdAt: json['createdAt'],
  );

  Map<String, dynamic> toJson() => {
    'name': name,
    'aliasName': aliasName,
    'pinyin': pinyin,
    'stockG': stockG,
    'unit': unit,
    'warnThresholdG': warnThresholdG,
  };

  bool get isLowStock => stockG <= warnThresholdG;
}

File: src/main/java/com/clinichub/entity/Herb.java
package com.clinichub.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "herb")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Herb {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(name = "alias_name", length = 100)
    private String aliasName;

    @Column(length = 100)
    private String pinyin;

    @Column(name = "stock_g", nullable = false)
    private Integer stockG;

    @Column(length = 10, nullable = false)
    private String unit;

    @Column(name = "warn_threshold_g", nullable = false)
    private Integer warnThresholdG;

    @Column(nullable = false)
    private Boolean enabled = true;

    @Column(name = "created_at", nullable = false)
    private java.time.LocalDateTime createdAt;
}
